src/wstool/multiproject_cmd.py

```diff
@@ -274,6 +276,32 @@ def do_work(self):
     return outputs
 
 
+def cmd_foreach(config, command, shell=False):
+    """Run command in all SCM entries in config, relative to path"""
+
+    class ForeachRetriever(object):
+        def __init__(self, element, command, shell):
+            self.element = element
+            self.command = command
+            self.shell = shell
+
+        def do_work(self):
+            command = self.command
+            if not self.shell:
+                command = shlex.split(command)
+            po = Popen(command, cwd=self.element.path, shell=self.shell,
```

Comment:
I believe the additional newlines come from a small bug in  vcstools:common.py._read_shell_output(), where print is being used instead of sys.out.write().